### PR TITLE
Feat 349 implement widget accident count by car type

### PR DESCRIPTION
**Pull request by 9Syncopation**
currently API in PR ,
working with mock data

### CODE REVIEW

File: src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  roadNumber: {
+    padding: theme.spacing(2),
+    alignSelf: 'flex-start',
+  },
+  textHighlight: {
+    backgroundColor: highlightBasicColor,
+    textAlign: 'center',
+  },
+}));
+// debugger
```
Comment (danielsh28) [inline src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx:28]: please remove debugger commented statement

File: src/components/molecules/DoubleBarChartView.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+interface IProps {
+  data: Array<object>;
+  yLabel: string | number;
+  xLabel1: string | number;
+  xLabel2: string | number;
+  textLabel: string;
+}
+const DoubleBarChartView: FC<IProps> = ({ data, xLabel1, xLabel2, yLabel }) => {
+  console.log('data double bar', data);
```
Comment (SejoB) [inline src/components/molecules/DoubleBarChartView.tsx:14]: remove console log

File: src/components/molecules/DoubleBarChartView.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// import React, { FC } from 'react';
```
Comment (danielsh28) [inline src/components/molecules/DoubleBarChartView.tsx:1]: remove comment

File: src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          <RoadNumberImage roadNumber={roadNumber} />
+        </div>
+      </Box>
+      {/* //TODO currently text is not recieved from API - need to fix */}
+      <DoubleBarChartView
+        data={items}
+        yLabel={CAR_TYPE}
+        xLabel1={PERCENTAGE_SEGMENT}
+        xLabel2={PERCENTAGE_COUNTRY}
+        textLabel={text?.title}
```
Comment (danielsh28) [inline src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx:50]: is text is optional? if so, then add `?` next to it in the `IWidgetAccidentCountByCarType`  interface definition in src/models/WidgetData.ts:
```
        text? : {
    title: string;
  };
```

File: src/components/molecules/DoubleBarChartView.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React, { FC } from 'react';
+import { ResponsiveContainer, BarChart, LabelList, XAxis, Bar } from 'recharts';
+import { DoubleBarChartFillColorBar1, DoubleBarChartFillColorBar2 } from '../../style';
+
+interface IProps {
+  data: Array<object>;
+  yLabel: string | number;
+  xLabel1: string | number;
```
Comment (yuvalbl) [inline src/components/molecules/DoubleBarChartView.tsx:8]: why `string | number` ?

File: src/components/molecules/DoubleBarChartView.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  xLabel2: string | number;
+  textLabel?: string;
+}
+const DoubleBarChartView: FC<IProps> = ({ data, xLabel1, xLabel2, yLabel }) => {
+  return (
+    <ResponsiveContainer width={'100%'} height={'70%'}>
```
Comment (yuvalbl) [inline src/components/molecules/DoubleBarChartView.tsx:14]: If possible, make container take the entire space, and determine the size from outside the component.
If not - `height` should be provided as a `prop`

File: src/components/molecules/DoubleBarChartView.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      <BarChart data={data} margin={{ top: 20, bottom: 15 }}>
```
Comment (yuvalbl) [inline src/components/molecules/DoubleBarChartView.tsx:15]: avoid adding inline-margin (or any style)
instead, use `<Box>`

File: src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    textAlign: 'center',
+  },
+}));
+const AccidentCountByCarType: FC<IProps> = ({ data, roadNumber, segmentText }) => {
+  const classes = useStyle();
+  const { items, text } = data;
+  return (
+    <>
+      <Box display="flex" flexDirection="column">
+        <span>השוואת אחוז התאונות ע"פ כלי רכב</span>
```
Comment (yuvalbl) [inline src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx:34]: Text should  use `i18n` (can be separate issue)

File: src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        <span>במקטע לעומת ממוצע ארצי בכבישים מאותו הסוג</span>
+        <span className={classes.textHighlight}>{segmentText}</span>
+      </Box>
+      <Box flexBasis={100} display="flex" flexDirection="column" justifyContent="center" alignItems="center">
+        <div className={classes.roadNumber}>
+          <RoadNumberImage roadNumber={roadNumber} />
+        </div>
+      </Box>
+      {/* //TODO currently text is not recieved from API - need to fix */}
```
Comment (yuvalbl) [inline src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx:43]: I think API already provide this data by now. If not - open and issue for it

File: src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx (hunk 4 above)
Comment (yuvalbl) [reply src/components/molecules/widgets/AccidentCountByCarTypeWidget.tsx:49]: I believe `text` is mandatory but `title` is not.
@9Syncopation why did you decide that? did you confirm it with @atalyaalon ?

**Review by yuvalbl — APPROVED** [review]

**Pull request merged**